src/bin/cargo-flamegraph.rs

```diff
@@ -269,6 +269,58 @@ impl std::fmt::Display for BinaryTarget {
     }
 }
 
+pub fn find_crate_root(manifest_path: Option<&Path>) -> anyhow::Result<PathBuf> {
+    match manifest_path {
+        Some(path) => {
+            let path = path.parent().ok_or_else(|| {
+                anyhow!(
+                    "the manifest path '{}' must point to a Cargo.toml file",
+                    path.display()
+                )
+            })?;
+
+            path.canonicalize().with_context(|| {
+                anyhow!(
+                    "failed to canonicalize manifest parent directory '{}'\nHint: make sure your manifest path is exists and points to a Cargo.toml file",
+                    path.display()
+                )
+            })
+        }
+        None => {
+            // heavily inspired by https://github.com/rust-lang/cargo/blob/1f12b88497f96b67ffdd7b1d41df98f690091358/src/cargo/util/important_paths.rs#L6
+            let valid_cargo_toml_file_name = "Cargo.toml";
+            let invalid_cargo_toml_file_name = "cargo.toml";
```

Comment:
While it makes sense for Cargo to do this, I think we can skip the details of checking for `cargo.toml`?

Reply:
Sure, fixed it. I also removed the comment as the new version is now a pretty naive approach without much resemblance to the original source.